test/tbb/test_eh_thread.cpp

```diff
@@ -33,9 +33,24 @@
 
 // On Windows there is no real thread number limit beside available memory.
 // Therefore, the test for thread limit is unreasonable.
-//
-// Under ASAN current approach is not viable as it breaks the ASAN itself as well
-#if TBB_USE_EXCEPTIONS && !_WIN32 && !__ANDROID__ && !__TBB_USE_ADDRESS_SANITIZER
+#if TBB_USE_EXCEPTIONS && !_WIN32 && !__ANDROID__
+
+#include <sys/types.h>
+#include <sys/time.h>
+#include <sys/resource.h>
+
+void limitThreads(size_t limit)
+{
+    rlimit rlim;
+
+    int ret = getrlimit(RLIMIT_NPROC, &rlim);
+    CHECK_MESSAGE(0 == ret, "getrlimit has returned an error");
+
+    rlim.rlim_cur = rlim.rlim_max == (rlim_t)RLIM_INFINITY ? limit : utils::min(limit, rlim.rlim_max);
```

Comment:
Just a minor comment to improve readability:
```suggestion
    rlim.rlim_cur = (rlim.rlim_max == (rlim_t)RLIM_INFINITY) ? limit : utils::min(limit, rlim.rlim_max);
```